src/kubernetes.cc

```diff
@@ -1052,6 +1271,14 @@ void KubernetesUpdater::StartUpdater() {
     pod_watch_thread_ = std::thread([=]() {
       reader_.WatchPods(watched_node, cb);
     });
+    if (config().KubernetesServiceMetadata()) {
```

Comment:
This actually needs both `config().KubernetesClusterLevelMetadata() && config().KubernetesServiceMetadata()`.